### PR TITLE
Check if port is int and raise exception

### PR DESCRIPTION
**Pull request by jppgomes**
**Proposed changes**:
- issue #8309

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by sara-tagger** [comment]
Thanks for submitting a pull request 🚀 @ArjaanBuijk will take a look at it as soon as possible ✨

**Comment by ArjaanBuijk** [comment]
@ancalita ,
Would you be able to review this update?

File: rasa/core/brokers/pika.py [hunk 1]
```diff
@@ -169,6 +170,10 @@ async def _connect(self) -> aio_pika.RobustConnection:
         last_exception = None
         for _ in range(self._connection_attempts):
             try:
+
+                if not isinstance(self.port, int):
+                    raise RasaException("Port has to be a integer.")
```
Comment (ancalita) [inline rasa/core/brokers/pika.py:175]: I think the RasaException can be raised a bit earlier - for example in the constructor [L75](https://github.com/RasaHQ/rasa/blob/a0399622910cc2ed7d003553fc664a3489a48ecc/rasa/core/brokers/pika.py#L75) where you can wrap the integer casting in a try-except block:
```
try:
     self.port = int(port)
except ValueError as e:
     raise RasaException("Port could not be converted to integer.") from e
```
You can try this by adding a new unit test at `rasa/tests/core/test_broker.py` and reproducing the Sentry error behaviour to see if this try-except block catches it.

**Review by ancalita — CHANGES_REQUESTED** [review]
Thanks for your contribution! 🚀 
In addition to adding a unit test to reproduce the Sentry error which checks that the added code catches it and raises RasaException, please also add a changelog entry at `rasa/changelog` by following the README instructions at that location.

File: rasa/core/brokers/pika.py (hunk 1 above)
Comment (jppgomes) [reply rasa/core/brokers/pika.py]: ok, thanks for the tips, would it be something like this?

File: tests/core/test_broker.py [hunk 2]
```diff
@@ -310,6 +311,23 @@ async def test_no_pika_logs_if_no_debug_mode(caplog: LogCaptureFixture):
         for record in caplog.records
     )
 
+    broker = PikaEventBroker(
+        "host", "username", "password", retry_delay_in_seconds=1, connection_attempts=1
+    )
```
Comment (ancalita) [inline tests/core/test_broker.py:316]: I'm not exactly sure what these new lines are for 🤔 If you added by mistake, could you please remove?

File: tests/core/test_broker.py [hunk 3]
```diff
@@ -310,6 +311,23 @@ async def test_no_pika_logs_if_no_debug_mode(caplog: LogCaptureFixture):
...
+
+
+def test_raise_exception_port(monkeypatch: MonkeyPatch):
```
Comment (ancalita) [inline tests/core/test_broker.py:319]: If there is no need for MonkeyPatch, this argument can be removed.

File: tests/core/test_broker.py [hunk 4]
```diff
@@ -310,6 +311,23 @@ async def test_no_pika_logs_if_no_debug_mode(caplog: LogCaptureFixture):
...
+
+    with pytest.raises(RasaException):
+        PikaEventBroker(
```
Comment (ancalita) [inline tests/core/test_broker.py:322]: I'm aware you cannot see the full stacktrace from Sentry, not sure why it got clipped, but I would actually use exactly what gets called further up in the stacktrace: the EventBroker factory from [here](https://github.com/RasaHQ/rasa/blob/51185ee5e66404864239b567dd67b34f77579fce/rasa/core/brokers/broker.py#L20).
You can create an instance of EndpointConfig which you can pass as argument to the broker factory:
`EndpointConfig(username="username", password="password", type="pika", port="PORT")

File: rasa/core/brokers/pika.py (hunk 1 above)
Comment (ancalita) [reply rasa/core/brokers/pika.py]: That was fast ⚡ 
It's looking in the right direction - I would also add an extra assertion after pytest.raises:
``` 
with pytest.raises(RasaException) as e:
     # to do

assert "Port could not be converted to integer." in str(e.value)
```

File: tests/core/test_broker.py (hunk 4 above)
Comment (ancalita) [reply tests/core/test_broker.py:322]: In addition, note of caution - the EventBroker factory is async so you'll need to use the keyword `await` when calling it. As well as making the test async too.

File: changelog/8575.bugfix.md [hunk 5]
```diff
@@ -0,0 +1 @@
+Check if port is int and raise exception in file pika.py
```
Comment (ancalita) [inline changelog/8575.bugfix.md:1]: Nitpicking now:

- I think you used the PR id in the changelog, you should use the issue id which I think is 8309.
- I would make the text a bit more descriptive : "Added check in `PikaEventBroker` constructor: if port cannot be cast to integer, raise `RasaException`."

File: tests/core/test_broker.py (hunk 3 above)
Comment (jppgomes) [reply tests/core/test_broker.py]: yes.. sorry, fixed!!

File: tests/core/test_broker.py (hunk 4 above)
Comment (jppgomes) [reply tests/core/test_broker.py:322]: ok, i'll do this

Comment (jppgomes) [reply tests/core/test_broker.py]: @ancalita can you check if it is right now?

Comment (ancalita) [reply tests/core/test_broker.py]: @jppgomes Sure, you could also re-request review by clicking on the arrows-in-a-circle icon to the right of the picture avatar under Reviewers section (top right).  
Does the test pass?
Please also add this assertion check in the test:
```
with pytest.raises(RasaException) as e:
     ...

assert "Port could not be converted to integer." in str(e.value)
```

File: tests/core/test_broker.py (hunk 2 above)
Comment (ancalita) [reply tests/core/test_broker.py:316]: @jppgomes This also needs to be resolved.

**Comment by ancalita** [comment]
> [CLA assistant check](https://cla-assistant.io/RasaHQ/rasa?pullRequest=8575)
> Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/RasaHQ/rasa?pullRequest=8575) before we can accept your contribution.
> **1** out of **2** committers have signed the CLA.
> 
> ✅ jppgomes
> ❌ C1300283 Joao Pedro Gomes Cabral Ferreira
> **C1300283 Joao Pedro Gomes Cabral Ferreira** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).
> You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/RasaHQ/rasa?pullRequest=8575) it.

@jppgomes There appears to be an issue with the license agreement not signed - did you commit changes from different devices maybe? You need to make sure that you've set an email address on your machine - this is an useful [guide.](https://docs.github.com/en/github/setting-up-and-managing-your-github-user-account/setting-your-commit-email-address)
After this is done, you might need to redo the commits to attach the email (via rebase).

File: tests/core/test_broker.py [hunk 6]
```diff
@@ -22,6 +22,7 @@
 from rasa.core.brokers.kafka import KafkaEventBroker
 from rasa.core.brokers.pika import PikaEventBroker, DEFAULT_QUEUE_NAME
 from rasa.core.brokers.sql import SQLEventBroker
+from rasa.shared.exceptions import RasaException
```
Comment (ancalita) [inline tests/core/test_broker.py:25]: i just noticed there is actually an existing import from the same module in L27 - could you add the `, RasaException` to L27 and remove duplicate import in L25?

Comment (jppgomes) [reply tests/core/test_broker.py:25]: ok

**Review by ancalita — APPROVED** [review]
Looks good, thank you 💯

**Comment by ancalita** [comment]
Hey @jppgomes I see you've run into some flakey CI test failures, it's fine to restart the jobs: go to Details of any failed check and you'll see somewhere on the right side Re-run jobs, select Re-run all jobs and that will restart the process. If you have this issue again, let me know and I can have a look.

**Comment by ancalita** [comment]
Hi @jppgomes could you please update branch with the base branch, that's something i can't do - I'll monitor the CI tests and restart if necessary.

**Comment by jppgomes** [comment]
ok @ancalita

**Comment by jppgomes** [comment]
@ancalita  it looks like everything is fine now

**Pull request merged**